### DISCUSSION TITLE
What's Nuxt entry point?

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [nuxt/nuxt#26210] [Questions]
Hello! I need to import `reflect-metadata` in the entry point of the app (in a Vue or NestJS app it would be the main.ts).
I tried everything, a module, a client plugin, a client middleware, in the app.vue's script, in the nuxt.config.ts.

I keep getting:

`
Error: tsyringe requires a reflect polyfill. Please add 'import "reflect-metadata"' to the top of your entry point.
`

I'm using `tsyringe` for DI, TypeScript class decorators, etc.
(still has some doubts about the proper way to enable TS decorators in Nuxt, even though I managed to silence the error by adding `vite: {
    esbuild: {
      target: 'ESNext',
      tsconfigRaw: {
        compilerOptions: {
          experimentalDecorators: true
        }
      }
    }
  },`.
  
  Any info will be appreciated.

**Comment by zavierrodriguez** [comment]
Did you ever get a solution to this?  I'm trying to do the exact same thing.

**Reply by angelhdzmultimedia** [reply]
Sadly not.  Nuxt abstracts away the main.ts and plugins install in the Vue app instance internally. 

But I suspected (and A.I. confirmed it) that initializing `reflect-metadata`  in a plugin which name starts with `0-` so its the first plugin to be run, it's the best approach for logic like this.

I've made many discussions about typescript decorators and DI and got no answer, so I lost interest in using them in Nuxt.
I'm still using classes but using explicit injection or service locator.

**Reply by zavierrodriguez** [reply]
Thanks for the update!  I just refactored my code to use direct dependency injection via constructors and it made the testing a breeze.  My application isn't huge yet so hopefully it doesn't become a huge burden as the app grows but for now it works!  Thanks again.
